frontend/scripts/buildCache.ts

```diff
@@ -51,36 +51,60 @@ async function testAPIAccess() {
   }
 }
 
-function cacheRPCensus() {
-  return seq(...DATASETS.map(_ => async () => {
-    console.log('  Getting RPs and Census for ' + safeDatasetHint(_))
-    let representativePoints = await getRepresentativePoints({ service_area_ids: _.serviceAreaIds })
-    let census = await getCensusData({ service_area_ids: _.serviceAreaIds })
+function cacheData() {
+  return seq(...DATASETS.map(dataset => async () => {
+    console.log('  Getting Rps and Census for ' + safeDatasetHint(dataset))
+    let representativePoints = await getRepresentativePoints({ service_area_ids: dataset.serviceAreaIds })
+    let census = await getCensusData({ service_area_ids: dataset.serviceAreaIds })
+    let storeLikeRps = representativePoints.map(_ => ({
+      ..._,
+      population: Number(_.population),
+      serviceAreaId: _.service_area_id,
+      demographics: census[_.service_area_id]
+    }))
+
     if (uploadToS3) {
-      let rpParams = { Bucket: s3Bucket, Key: getS3Key(getStaticRPUrl(_)), Body: JSON.stringify(representativePoints), ContentType: 'application/json', ACL: 'public-read' }
-      let censusParams = { Bucket: s3Bucket, Key: getS3Key(getStaticDemographicsUrl(_)), Body: JSON.stringify(census), ContentType: 'application/json', ACL: 'public-read' }
+      let rpParams = { Bucket: s3Bucket, Key: getS3Key(getStaticRPUrl(dataset)), Body: JSON.stringify(representativePoints), ContentType: 'application/json', ACL: 'public-read' }
+      let censusParams = { Bucket: s3Bucket, Key: getS3Key(getStaticDemographicsUrl(dataset)), Body: JSON.stringify(census), ContentType: 'application/json', ACL: 'public-read' }
       s3.putObject(rpParams, s3Callback)
       s3.putObject(censusParams, s3Callback)
     }
-  }))
-}
 
-function cacheAdequacies() {
-  return seq(...DATASETS.map(dataset => async () =>
     seq(...METHODS.map(method => async () => {
-      console.log('  Getting Adequacy for ' + safeDatasetHint(dataset) + ' for ' + method)
+      console.log('  Getting Adequacy and result CSV for ' + safeDatasetHint(dataset) + ' for ' + method)
       let adequacies = await getAdequacies({
         method,
         providers: dataset.providers.map((_, n) => ({ latitude: _.lat, longitude: _.lng, id: n })),
         service_area_ids: dataset.serviceAreaIds,
         dataset_hint: safeDatasetHint(dataset)
       })
-      if (uploadToS3 && adequacies) {
+
+      let hash = keyBy(storeLikeRps, 'id')
```

Comment:
seems like this function is getting a bit long, does it make sense to break it up more? Is it also possible to add some comments to say what's going on in the different parts?